.github/workflows/build.yaml

```diff
@@ -158,33 +158,63 @@ jobs:
       matrix:
         kind-k8s-version: [1.22.17, 1.23.17, 1.24.13, 1.25.9, 1.26.4, 1.27.2]
         vault-version: [1.11.10, 1.12.6, 1.13.2]
-        # Note: The below exclude and includes are a bit awkward, but they
-        # allow us to exclude the combos we don't really need. We want to test
-        # the operator with the different k8s versions, and with the different
-        # vault versions, but we don't care about testing all the k8s versions
-        # against all the vault versions.
+        installation-method: [helm, kustomize]
+        software-type: [ent, oss]
+        # Note: We want to test the operator with the different k8s versions,
+        # and with the different vault versions, but we don't care about testing
+        # all the k8s versions against all the vault versions.
         # Combos to exclude:
         #   kind-k8s-version: [1.22.17, 1.23.17, 1.24.13, 1.25.9, 1.26.4]
         #   vault-version: [1.11.10, 1.12.6]
+        # We run kustomize ent against the full matrix, and other tests against vault version 1.13.2.
+        # Combos to exclude:
+        #   vault-version: [1.11.10, 1.12.6]
+        #   any combos from installation-method and software-type except for kustomize ent
+        # TODO: When the VDS integration test supports Helm, test kustomize ent against one version of (ENT) Vault like the other tests.
+        # TODO: When the VDS integration test supports Helm, swap the matrix filter so helm ent test runs the full matrix.
         exclude:
+          # vault version 1.11.10
           - kind-k8s-version: 1.26.4
+            vault-version: 1.11.10
           - kind-k8s-version: 1.25.9
+            vault-version: 1.11.10
           - kind-k8s-version: 1.24.13
+            vault-version: 1.11.10
           - kind-k8s-version: 1.23.17
+            vault-version: 1.11.10
           - kind-k8s-version: 1.22.17
-        include:
+            vault-version: 1.11.10
+          - vault-version: 1.11.10
+            installation-method: helm
+            software-type: ent
```

Comment:
nit: could we maybe rename this to `enterprise` and have its value be a boolean e.g: `true`. That would more closely align with what have in the Makefile.